### PR TITLE
Add registration page

### LINKED ISSUE #12: Usernames are case sensitive
Currently, we query for exact match of username. This should be changed to a case-insensitive match for the username so that e.g. `Cypress`/`abc123` works as well as `cypress`/`abc123`. This also affects registration -- we don't want two people with the same username just different capitalisation.

### LINKED ISSUE #15: Registration page
Users need to be able to sign up, this needs to check the list of allowed emails.

~Currently blocked on #16~ — issue now closed

### LINKED ISSUE #28: Use hasura for authentication
We currently connect directly to the database for authentication, but we could probably tidy up how this works by connecting via Hasura. In order not to expose salted passwords over the API, we would need a special server role to retrieve the information (or the server could use the admin secret, but that's probably not ideal security-wise).﻿

### LINKED ISSUE #55: Deploy script pulls wrong tag
Currently `deploy.sh` pulls `sapper:latest`, but `docker-compose.prod.yml` uses `sapper:slim`. This means that the site doesn't deploy the new version properly.

To fix this, `deploy.sh` should instead use something like `docker-compose -f docker-compose.prod.yml pull` so it always stays in sync with the compose file. Additionally, we should run shellcheck on the deploy file (we could also add it as a lint step in CI/via husky, but that sounds like it might be overkill).

### PR DESCRIPTION
**Pull request by jarhodes314**
Fixes #55 
Fixes #12 
Fixes #15
Fixes #28

### CODE REVIEW

File: deploy.sh [hunk 1]
```diff
@@ -2,10 +2,11 @@
 set -e # Fail script if a command fails
 
 # Start the ssh agent (required to clone/pull)
-eval $(ssh-agent -s)
+eval "$(ssh-agent -s)"
 ssh-add <(cat ~/.ssh/deploy_pull)
 
 # Load environment variables passed in from CI container
+# shellcheck source=/dev/null
```
Comment (caro401) [inline deploy.sh:9]: what does that mean?

File: hasura/migrations/1612475310822_create_table_cucb_list042/up.sql [hunk 2]
```diff
@@ -0,0 +1 @@
+CREATE TABLE "cucb"."list042"("email" text NOT NULL, PRIMARY KEY ("email") ); COMMENT ON TABLE "cucb"."list042" IS E'List of email addresses signed up to the main list';
```
Comment (caro401) [inline hasura/migrations/1612475310822_create_table_cucb_list042/up.sql:1]: does there need to be docs about what/why list042?

File: src/auth.js [hunk 3]
```diff
@@ -1,44 +1,174 @@
...
+      } else {
+        throw errors.INTERNAL_ERROR;
+      }
+    }
+  } else {
+    throw errors.NOT_ON_MAILING_LIST;
+  }
+};
+
+function mysql_real_escape_string (str) {
```
Comment (caro401) [inline src/auth.js:149]: what/why? not using mysql anymore?

File: src/routes/auth/register.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
+import { createAccount } from "../../auth";
+export const EMAIL_PATTERN = /^(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])$/;
```
Comment (caro401) [inline src/routes/auth/register.js:2]: is this nicked from somewhere on the internet? can you document where?

File: src/routes/auth/register.svelte [hunk 5]
```diff
@@ -0,0 +1,138 @@
+<script>
+  import { createValidityChecker, makeTitle, themeName, committee } from "../../view";
+  import Mailto from "../../components/Mailto.svelte";
+  // TODO stick these somewhere to be re-used
```
Comment (caro401) [inline src/routes/auth/register.svelte:4]: do that thing! nasty regex is much nastier when its duplicated

File: src/view.js [hunk 6]
```diff
@@ -24,3 +24,79 @@ export const themeName = writable("");
 
 export const suffix = (n) =>
   ({ one: "st", two: "nd", few: "rd", other: "th" }[new Intl.PluralRules("en-gb", { type: "ordinal" }).select(n)]);
+
+export const createValidityChecker = () => {
```
Comment (caro401) [inline src/view.js:28]: could the icky regex live here maybe?

**Review by caro401 — DISMISSED** [review]

File: deploy.sh (hunk 1 above)
Comment (jarhodes314) [reply deploy.sh:9]: https://github.com/koalaman/shellcheck/wiki/SC1090 it can't know what variables are set by it. I don't care so used /dev/null

**Pull request merged**